contracts/Remittance.sol

```diff
@@ -0,0 +1,38 @@
+pragma solidity >=0.4.21 <0.6.0;
+
+import "./Pausable.sol";
+
+contract Remittance is Pausable {
+  address public owner;
+  address public carol;
+  bytes32 private withdrawHash;
+
+  event LogDeposited(address indexed sender, uint256 amount, bytes32 hash, address indexed carol);
+  event LogWithdrawn(address indexed sender, uint256 amount);
+
+  constructor(bool paused) Pausable(paused) public {
+  }
+
+  function calcuateHash(address _carol, bytes32 _bobPassword, bytes32 _carolPassword) public view returns(bytes32) {
+     return keccak256(abi.encodePacked(this, _carol, _bobPassword, _carolPassword));
```

Comment:
Two instances of the same contract will yield the same hashes. How can you protect your users from previously disclosed passwords?

Reply:
Sorry, I don't get this here, shall Alice always use different passwords for Bob and Carol every time to avoid getting the same `withdrawHash` again since the `withdrawHash`  can be found on-chain?

Because we need to re-call this function again, if we use something like `block.timestamp` we aren't able to get the same hash when `withdraw`. Any better way to generate different hash event providing the same passwords again?

Reply:
If you use the `block.timestamp`, you better be sure that the tx is mined at the right time. And even then, it is impossible.
Think of another thing that is unique per deployed contract.

Reply:
is `this` is enough here? or I need to use `address(this)` to use the contract address which won't change for a deployed contract.

Reply:
It is enough indeed. Both are equivalent. New compilers complain on a naked `this`.